controllers/volumereplicationgroup_controller.go

```diff
@@ -312,8 +425,84 @@ func (v *VolumeReplicationGroupReconciler) HandlePersistentVolumes(
 	return nil
 }
 
+// HandlePersistentVolumes handles bound PVs
+func (v *VolumeReplicationGroupReconciler) HandlePersistentVolumes(
+	ctx context.Context,
+	pvcList *corev1.PersistentVolumeClaimList) bool {
+	requeue := false
+
+	if err := v.PrintPersistentVolumes(ctx, pvcList); err != nil {
+		log.Error(err, "failed to print the persistent volumes for pvc list %w")
+
+		requeue = true
+	}
+
+	return requeue
+}
+
+func (v *VolumeReplicationGroupReconciler) CreateVolumeReplicationCRs(
+	ctx context.Context,
+	pvcList *corev1.PersistentVolumeClaimList) bool {
+	requeue := false
+
+	for _, pvc := range pvcList.Items {
+		// if the PVC is not bound yet, dont proceed.
+		if pvc.Status.Phase != corev1.ClaimBound {
+			log.Errorf("PVC is not yet bound status: %v", pvc.Status.Phase)
+
+			requeue = true
+
+			return requeue
+		}
+
+		volRep := &volrep.VolumeReplication{}
+
+		err := v.Get(ctx, types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, volRep)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				if err = v.CreateVolumeReplicationCRForPVC(ctx, pvc); err == nil {
+					continue
+				}
+			}
+
+			// requeue on failure to ensure any PVC not having a corresponding VR CR
+			requeue = true
+
+			log.Error(err, "failed to get or create VolumeReplication CR for PVC ", pvc.Name, pvc.Namespace)
+		}
+	}
+
+	return requeue
+}
+
 // createVolumeReplicationCRForPVC creates a VolumeReplication CR with a PVC as its data source
-func (v *VolumeReplicationGroupReconciler) createVolumeReplicationCRForPVC(
+func (v *VolumeReplicationGroupReconciler) HandlePersistentVolumeClaims(
+	ctx context.Context,
+	volRepGroup *ramendrv1alpha1.VolumeReplicationGroup,
+	pvcList *corev1.PersistentVolumeClaimList) bool {
+	// HandlePersistentVolumeClaims creates VolumeReplication CR for each pvc
+	// from pvcList. If fails (even for one pvc), then requeue is set to true.
+	// For now, keeping creation of VolumeReplication CR and backing up of PV
+	// metadata in separate functions and calling them separately. In future,
+	// if creation of VolumeReplication CR and backing up of pv metadata should
+	// happen together (i.e for each pvc create VolumeReplication CR and then
+	// backup PV metadata), then it can be put in a single function.
+	if requeue := v.CreateVolumeReplicationCRs(ctx, pvcList); requeue {
+		log.Error("failed to get or create VolumeReplication CRs for pvc list")
+
+		return requeue
```

Comment:
(future) as the comment above states. We could possibly create VR and backup the PV, per PVC instead of waiting for all VRs to be created before backing the PVs up.

Reply:
Yes. This is deferred to a separate PR.